### ISSUE TITLE
`my sub` should be parsed as `PPI::Statement::Sub`

### ISSUE THREAD

**Issue by gugod** [adamkennedy/PPI#227]
PPI 1.236 parses `my sub NAME { ... }` as  `PPI::Statement::Variable` object instead of `PPI::Statement::Sub` object.

```
> cat -n x.pl
     1  my sub bar {
     2      return 1;
     3  }
     4

> perl -MPPI::Document -MPPI::Dumper -e 'PPI::Dumper->new(PPI::Document->new($ARGV[0]), whitespace => 0)->print' x.pl
PPI::Document
  PPI::Statement::Variable
    PPI::Token::Word    'my'
    PPI::Token::Word    'sub'
    PPI::Token::Word    'bar'
    PPI::Structure::Block       { ... }
      PPI::Statement::Break
        PPI::Token::Word        'return'
        PPI::Token::Number      '1'
        PPI::Token::Structure   ';'
```

`state sub NAME { ...  }` is also parsed the same way.

It would be much more convienent if they are parsed as `PPI::Statement::Sub` object.

**Comment by akiym** [comment]
It will be fixed by #210.

**Comment by wchristian** [comment]
This was fixed by #210 and is now on cpan.
